pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -0,0 +1,776 @@
+                            .whenComplete((v, e) -> {
+                                        if (e != null) {
+                                            getOwnerRequests.remove(serviceUnit);
+                                            log.warn("Failed to getOwner for serviceUnit:{}",
+                                                    serviceUnit, e);
+                                            future.completeExceptionally((Throwable) e);
+                                        }
+                                    }
+                            );
+                    return future;
+                });
+    }
+
+    private CompletableFuture<Integer> closeServiceUnit(String serviceUnit) {
+        long startTime = System.nanoTime();
+        MutableInt unloadedTopics = new MutableInt();
+        NamespaceBundle bundle = getNamespaceBundle(serviceUnit);
+        return pulsar.getBrokerService().unloadServiceUnit(
+                        bundle,
+                        false,
+                        pulsar.getConfig().getNamespaceBundleUnloadingTimeoutMs(),
+                        TimeUnit.MILLISECONDS)
+                .handle((numUnloadedTopics, ex) -> {
+                    if (ex != null) {
+                        // ignore topic-close failure to unload bundle
+                        log.error("Failed to close topics under bundle:{}",
+                                bundle.toString(), ex);
+                    } else {
+                        unloadedTopics.setValue(numUnloadedTopics);
+                    }
+                    // clean up topics that failed to unload from the broker ownership cache
+                    pulsar.getBrokerService().cleanUnloadedTopicFromCache(bundle);
+                    return numUnloadedTopics;
+                })
+                .whenComplete((ignored, ex) -> {
+                    double unloadBundleTime = TimeUnit.NANOSECONDS
+                            .toMillis((System.nanoTime() - startTime));
+                    log.info("Unloading bundle:{} with {} topics completed in {} ms",
+                            bundle, unloadedTopics, unloadBundleTime, ex);
+                });
+    }
+
+    private CompletableFuture<Void> splitServiceUnit(String serviceUnit) {
+        // TODO: after the split we need to write the child ownerships to BSC instead of ZK.
+        //  Use the configured NamespaceBundleSplitAlgorithm
+        long startTime = System.nanoTime();
+        return pulsar.getNamespaceService()
+                .splitAndOwnBundle(getNamespaceBundle(serviceUnit),
+                        false,
+                        NamespaceBundleSplitAlgorithm.RANGE_EQUALLY_DIVIDE_ALGO,
+                        null)
+                .whenComplete((__, ex) -> {
+                    double splitBundleTime = TimeUnit.NANOSECONDS
+                            .toMillis((System.nanoTime() - startTime));
+                    log.info("Splitting {} namespace-bundle completed in {} ms",
```

Comment:
we should log it as ERROR and not "info" in case ex is not null (and log the error)

Reply:
updated.